### PR TITLE
Fix: Set updating is broken

### LINKED ISSUE #32: Set updating is broken
In september 2019, the Koordinates API made a backward incompatible change to add versioning to sets. This removed the PUT method for `/sets/:id/`, and added endpoints for `versions/` in the same way that layers etc behave.

This client has yet to be updated to match (PUT for the set-detail endpoint is [still present](https://github.com/koordinates/python-client/blob/master/koordinates/client.py#L257-L259) in the code)

### PR DESCRIPTION
**Pull request by agonzalezv**
GIF

### ⚠️ Issue/Task Card

closes #32 

### 👎 Existing Behaviour

- Deprecated PUT method for `/sets/:id/` is present in the code

### 👍 Expected Behaviour Now

- The opposite of above

### 🚀 Types of changes

What kind of change is this? 

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

### ✅ Checklist:

- [x] Code changes.
- [x] Specs have been written and pass.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by agonzalezv** [comment]
@hamishcampbell @craigds I'm not 100% about my approach to unit tests, it feels a bit too simplistic. I appreciate your input/feedback in that if you have any.

File: koordinates/sets.py [hunk 1]
```diff
@@ -74,27 +145,204 @@ def _deserialize(self, data, manager):
...
+            version_id = self.version.id
+
+        target_url = self._client.get_url(
+            "SET_VERSION",
+            "DELETE",
+            "single",
+            {"id": self.id, "version_id": version_id},
+        )
+        r = self._client.request("DELETE", target_url)
+        logger.info("delete_version(): %s", r.status_code)
```
Comment (craigds) [inline koordinates/sets.py:322]: client.request already has logging for this stuff, so not sure this is necessary

File: koordinates/sets.py [hunk 2]
```diff
@@ -74,27 +145,204 @@ def _deserialize(self, data, manager):
...
+            "SET_VERSION", "POST", "publish", {"id": self.id, "version_id": version_id},
+        )
+        r = self._client.request("POST", target_url, json={})
+        return self._client.get_manager(Publish).create_from_result(r.json())
+
+    @is_bound
+    def save(self, with_data=False):
+        """
+        Edits this draft layerversion.
+        # If anything in the data object has changed, cancel any existing import and start a new one.
```
Comment (craigds) [inline koordinates/sets.py:287]: as before, I think the 'data object' stuff here is wrong, sets don't have that stuff

File: koordinates/sets.py [hunk 3]
```diff
@@ -74,27 +145,204 @@ def _deserialize(self, data, manager):
...
+            "SET_VERSION", "GET", "published", {"id": self.id}
+        )
+        return self._manager._get(target_url, expand=expand)
+
+    @is_bound
+    def create_draft_version(self):
+        """
+        Creates a new draft version from this model content.
+
+        If anything in the data object has changed then an import will begin immediately.
```
[the remark this answers is not in the document]
Comment (craigds) [reply koordinates/sets.py:252]: I'm not super experienced with this code, but it looks like there's both a `SetManager.create_draft` _and_ a `Set.create_draft_version` method? Neither calls the other. Is that intentional, and could we de-duplicate them easily? Does the same exist for other models?

**Comment by craigds** [comment]
> @hamishcampbell @craigds I'm not 100% about my approach to unit tests, it feels a bit too simplistic. I appreciate your input/feedback in that if you have any.

Simple is good! I actually think they look pretty good...

File: koordinates/sets.py [hunk 4]
```diff
@@ -42,6 +52,67 @@ def create(self, set):
         r = self.client.request("POST", target_url, json=set._serialize())
         return set._deserialize(r.json(), self)
 
+    def list_versions(self, set_id):
+        """
+        Filterable list of versions of a set, always ordered newest to oldest.
+
+        If the version’s source supports revisions, you can get a specific revision using
+        ``.filter(data__source__revision=value)``. Specific values depend on the source type.
```
[the remark this answers is not in the document]
Comment (agonzalezv) [reply koordinates/sets.py:60]: what's the difference between `data__source_revision` and `data__source__revision` ? is it a typo, or they are 2 different things?

File: koordinates/sets.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (agonzalezv) [reply koordinates/sets.py:252]: layers has both `create_draft_version` : https://github.com/koordinates/python-client/blob/2944e15d64d1a66dad7a632773867fc922dcf459/koordinates/layers.py#L305 

and 

`create_draft` : https://github.com/koordinates/python-client/blob/2944e15d64d1a66dad7a632773867fc922dcf459/koordinates/layers.py#L109

but that seem to be related with the import. I'll tidy up the code in `sets`.

File: koordinates/sets.py [hunk 5]
```diff
@@ -42,6 +52,67 @@ def create(self, set):
...
+        target_url = self.client.get_url(
+            "SET_VERSION", "GET", "published", {"id": set_id}
+        )
+        return self._get(target_url, expand=expand)
+
+    def create_draft(self, set_id):
+        """
+        Creates a new draft version.
+
+        If anything in the data object has changed then an import will begin immediately.
```
Comment (craigds) [inline koordinates/sets.py:103]: missed this one

Comment (agonzalezv) [reply koordinates/sets.py]: fixed

**Pull request merged**